### ISSUE TITLE
Spike: Investigating End-to-End (E2E) and Visual Regression Testing

### ISSUE THREAD

**Issue by moathabuhamad-cengage** [cengage/react-magma#1550]
#### Background:
We currently require manual testing of each component before release.
Automating tests will streamline this process, improve consistency, and catch issues earlier.

#### Initial Thoughts
- **Playwright**: Modern, supports Chromium, Firefox, WebKit, and easy TypeScript setup.
- **Argos CI**: Free for open-source, integrates with GitHub, great for visual testing.

#### Investigation
- Review testing strategies of:
  - **MUI**.
  - **IBM Carbon**.
  - **Adobe Spectrum**.

**Comment by olehnoskov-cengage** [comment]
Comparison of some of the popular UI E2E testing tools:

1. **Selenium**
2.

**Comment by silvalaura** [comment]
@olehnoskov-cengage What major differences do you see between Playwright and Cypress? What are pros/cons of choosing one over the other? How easy or hard would it be to get started with either of those?

**Comment by silvalaura** [comment]
I agree with moving forward with Playwright. @olehnoskov-cengage can you create one or two new tickets capturing the work? maybe one to get the project set up with minimal examples, and then a few others for testing specific components? then I think we can close this.

**Comment by olehnoskov-cengage** [comment]
Links for created tickets:

https://github.com/cengage/react-magma/issues/1726

https://github.com/cengage/react-magma/issues/1727
